File: GCD.java
import java.util.*;
 public class GCD{  
    //for two number
    static int gcd(int a,int b){ 
        return b==0 ? a : gcd(b, a%b);
    }  
//for n number
    static int gcdOfNNum(int a[], int n){  
        int ans =0; 
        for(int i=0; i<n; i++){ 
            ans = gcd(ans,a[i]);
        } 
        if(ans == 1){ 
            return 1;
        } 
        return ans;

    }
    public static void main(String[]args){   
        Scanner sc = new Scanner(System.in); 
        int t = sc.nextInt(); 
        int n = sc.nextInt();  
        int a[] = new int[n];
 
        for(int i=0; i<n; i++){ 
            a[i]=sc.nextInt();
        }
        // int a[]={1,2,3};  
        // int n = a.length; 
        
       // System.out.println( gcdOfNNum(a,n)); 
       int result = gcdOfNNum(a,n); 
       System.out.println(result);
}
} 

File: GCD.cpp
#include<iostream> 
using namespace std; 
 int GCD(int a,int b){ 
    if(a==0) {
    
    return b; 
    } 
    if(b==0){ 
     
        return a;
    } 
    while(a!=b){ 
        if(a>b){ 
            a=a-b;
        } 
        else{
        b=b-a; 
        }
    } 
return a;
} 
int main(){ 
    int a,b; 
    cout<<"enter the vale a and b :"<<endl; 
    cin>>a>>b; 
int ans=GCD(a,b); 
cout<<" the gcd of :"<<a<<" & "<<b<<" is :"<<ans<<endl; 
 
return 0;
}